tests/callbacks/test_kill_loss_spike_callback.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2024 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+import unittest
+from collections import deque
+from unittest.mock import patch
+
+from llmfoundry.callbacks.kill_loss_spike_callback import KillLossSpike
+
+
+class TestKillLossSpike(unittest.TestCase):
```

Comment:
can you also test that setting log_only works properly? i.e. the default settings won't kill a run

Reply:
Yep I have manually tested this by submitting runs with `kill_loss_spike: {log_only: true}`. For example: [https://dbc-559ffd80-2bfc.cloud.databricks.com/ml/experiments/2436561715528145/runs/43cc2a5856e74d2985f554be2396ff8d/model-metrics?o=7395834863327820](url)

Reply:
can we unit test it too though?

Reply:
Decomposed the spike-handling and unit tested log_only setting